packages/rum-core/src/rawRumEvent.types.ts

```diff
@@ -209,3 +209,31 @@ export interface CommonContext {
   context: Context
   hasReplay?: true
 }
+
+export interface ReadonlyLocation {
+  readonly hash: string
+  readonly host: string
+  readonly hostname: string
+  readonly href: string
+  readonly origin: string
+  readonly pathname: string
+  readonly port: string
+  readonly protocol: string
+  readonly search: string
+}
+
+export type RumEventDomainContext<E extends RawRumEvent> = E extends RawRumErrorEvent
+  ? { error?: unknown }
+  : E extends RawRumActionEvent
+  ? { event?: Event }
+  : E extends RawRumLongTaskEvent
+  ? { performanceEntry: PerformanceEntry }
+  : E extends RawRumResourceEvent
+  ? {
+      performanceEntry?: PerformanceEntry
```

Comment:
Some of them are not actual `PerformenceEntry` objects, for example on `initial_document` and XHR/fetch requests. I wonder if it would be worth it to implement a proper type for it?